src/routes/users/-components/UsersList.tsx

```diff
@@ -1,20 +1,17 @@
-import { Fragment } from 'react';
-import { useSearchParams } from 'react-router-dom';
+import { useSearch, useNavigate } from '@tanstack/react-router';
 
 import { CodeBlock } from 'ui/codeBlock/CodeBlock';
 import { useQuery } from 'hooks/useQuery/useQuery';
 
 type SortType = 'asc' | 'desc';
 
 export const UsersList = () => {
-  const [searchParams, setSearchParams] = useSearchParams();
-
-  const sort = searchParams.get('sort') as SortType;
-  const page = searchParams.get('page');
+  const { sort, page } = useSearch({ from: '/users/' });
+  const navigate = useNavigate();
 
   const { data: usersResponse, isFetched: areUsersFetched } = useQuery(
     'getUsersList',
-    { page: page || undefined },
+    { page: page?.toString() || undefined },
```

Comment:
I think it can be simplified
```suggestion
    { page: page?.toString() },
```